src/tactic/move_add.lean

```diff
@@ -0,0 +1,245 @@
+/-
+Copyright (c) 2022 Arthur Paulino, Damiano Testa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Arthur Paulino, Damiano Testa
+-/
+import tactic.core
+
+/-!
+# `move_add`: a tactic for moving summands
+
+Calling `move_add [a, ← b, c]`, recursively looks inside the goal for expressions involving a sum.
+Whenever it finds one, it moves the terms `a, b, c`, removing all parentheses.  The terms preceded
+by a `←` get placed to the left, the ones without the arrow get placed to the right.  Unnamed terms
+stay in place.  Due to re-parenthesizing, doing `move_add` with no argument may change the goal.
+Also, the *order* in which the terms are provided matters: the tactic reads them from left to right.
+This is especially important if there are multiple matches for the typed terms in the given
+expressions.
+
+### Warning:
+* The tactic uses lemmas from `algebra.group.basic`, thus it will not work unless this file is
+  imported
+* The tactic will discard user-provided terms that do not unify with something in the expression.
+  This means that the tactic will not give an error if it finds no match of the provided terms.
+  The reason for this choice is that we want a single call of `move_add` to move terms across
+  different sums in the same expression.  Here is an example.
```

Comment:
This rationale only applies if using the tactic in a combinator like `all_goals`, which does not appear to be the case in the examples. If the tactic is handling the iteration itself because it was passed a list of things to do, it report errors for non-matching terms without affecting the intended behavior.

Reply:
Ok, I will work more on understanding how to extract information from the various branches and how to report it and will try to update the tactic and the doc-module.

Reply:
To me, this "allow unmatching patterns" behavior feels like it should be hidden behind a flag.

Reply:
I am trying to catch the presence of unmatching patterns, as per [this discussion](https://leanprover.zulipchat.com/#narrow/stream/239415-metaprogramming-.2F-tactics/topic/catching.20errors.3F/near/280019039)

Reply:
The tactic now should report ununified variables and unmodified targets.  It currently treats `at *` as though it was the union of all hypotheses + goal in terms of error reporting, instead of simply making sure that *something* changes (and whathever the correct condition for non-unification should be).

I am taking a lunch break now!

Reply:
Error reporting for `at *` should be correct now (though I tested it in just a handful of cases).